clientlibrary/config/config.go

```diff
@@ -169,6 +170,17 @@ type (
 		// StreamName is the name of Kinesis stream
 		StreamName string
 
+		// EnableEnhancedFanOutConsumer enables enhanced fan-out consumer
+		// See: https://docs.aws.amazon.com/streams/latest/dev/enhanced-consumers.html
+		// Either consumer name or consumer ARN must be specified when Enhanced Fan-Out is enabled.
+		EnableEnhancedFanOutConsumer bool
+
+		// EnhancedFanOutConsumerName is the name of Kinesis Enhanced Fan-Out consumer.
```

Comment:
https://github.com/awslabs/amazon-kinesis-client/blob/cce05bf4fd03eed007bf6c51203f3cccc189c92a/amazon-kinesis-client/src/main/java/software/amazon/kinesis/retrieval/fanout/FanOutConfig.java#L51

Reply:
Should I use `ApplicationName` as `ConsumerName` if `ConsumerName` is not specified and Enhanced Fan-Out is enabled?

Reply:
yes, this will be in-line with aws java kcl design.